Mathlib/RingTheory/Coalgebra.lean

```diff
@@ -90,6 +92,63 @@ theorem counit_apply (r : R) : counit r = r := rfl
 
 end CommSemiring
 
+namespace Prod
+variable (R : Type u) (A : Type v) (B : Type w)
+variable [CommSemiring R] [AddCommMonoid A] [AddCommMonoid B] [Module R A] [Module R B]
+variable [Coalgebra R A] [Coalgebra R B]
+
+open LinearMap in
+instance instCoalgebra : Coalgebra R (A × B) where
+  comul := .coprod
+    (TensorProduct.map (.inl R A B) (.inl R A B) ∘ₗ comul)
+    (TensorProduct.map (.inr R A B) (.inr R A B) ∘ₗ comul)
+  counit := .coprod counit counit
+  rTensor_counit_comp_comul := by
+    ext x : 2 <;> dsimp
```

Comment:
This is a pair of non-terminal `dsimp`s. Shouldn't we squeeze them and just append the results to the `simp only`s?

Reply:
Same remark for the proofs of `lTensor_counit_comp_comul` and `coassoc` below.